addons/l10n_ch/models/res_bank.py

```diff
@@ -137,9 +138,22 @@ def build_swiss_code_url(self, amount, currency_name, not_used_anymore_1, debtor
             comment,                                              # Unstructured Message
             'EPD',                                                # Mandatory trailer part
         ]
+        return qr_code_vals
 
+    @api.model
+    def build_swiss_code_url(self, amount, currency_name, not_used_anymore_1, debtor_partner, not_used_anymore_2, structured_communication, free_communication):
+        qr_code_vals = self.build_swiss_code_vals(amount, currency_name, not_used_anymore_1, debtor_partner, not_used_anymore_2, structured_communication, free_communication)
         return '/report/barcode/?type=%s&value=%s&width=%s&height=%s&humanreadable=1' % ('QR_quiet', werkzeug.urls.url_quote_plus('\n'.join(qr_code_vals)), 256, 256)
 
+    @api.model
+    def build_swiss_code_base64(self, amount, currency_name, not_used_anymore_1, debtor_partner, not_used_anymore_2, structured_communication, free_communication):
+        qr_code_vals = self.build_swiss_code_vals(amount, currency_name, not_used_anymore_1, debtor_partner, not_used_anymore_2, structured_communication, free_communication)
+        try:
+            barcode = self.env['ir.actions.report'].barcode('QR_quiet', '\n'.join(qr_code_vals), width=256, height=256, humanreadable=1)
+        except (ValueError, AttributeError):
+            raise werkzeug.exceptions.HTTPException(description='Cannot convert into barcode.')
+
+        return image_data_uri(base64.b64encode(barcode))
```

Comment:
Arent't we going to face the same kind of problems with SEPA QR-codes then ? Shouldn't we do the same for them ?
https://github.com/odoo/odoo/blob/12.0/odoo/addons/base/models/res_bank.py#L131

(Note for forward-port: this had nothing to do in base, and got moved to a new account_qr_code_sepa module in v14)